### PR TITLE
FEAT - update and make loki production grade and test it with k8s 1.25

### PR DESCRIPTION
**Pull request by nutellinoit**
Removed Loki single and replaced with Loki distributed.

- Made Loki distributed production grade, by default every component has 1 replica and can autoscale to more, if needed.
- Removed non-distributed minio on logging operated to store errors logs
- Added minio-ha to store error logs and loki objects

### CODE REVIEW

File: docs/releases/v3.1.0.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# DRAFT Logging Core Module Release 3.1.0
+
+Welcome to the latest release of `logging` module of [`Kubernetes Fury Distribution`](https://github.com/sighupio/fury-distribution)
+maintained by team SIGHUP.
```
Comment (ralgozino) [inline docs/releases/v3.1.0.md:4]:
```suggestion
Welcome to the latest release of the `logging` module of [`Kubernetes Fury Distribution`](https://github.com/sighupio/fury-distribution) maintained by team SIGHUP.
```

File: docs/releases/v3.1.0.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+This update includes new versions of various components, such as OpenSearch and OpenSearch Dashboards,
+as well as bug fixes and changes, such as support for Kubernetes 1.25 and the introduction of a new Loki distributed package and minio-ha package.
```
Comment (ralgozino) [inline docs/releases/v3.1.0.md:7]:
```suggestion
This update includes new versions of various components, such as OpenSearch and OpenSearch Dashboards, as well as bug fixes and changes, such as support for Kubernetes 1.25 and the introduction of two new packages, Loki distributed and minio-ha.
```

File: docs/releases/v3.1.0.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+| `opensearch`             | [`v2.5.0`](https://github.com/opensearch-project/OpenSearch/releases/tag/2.5.0)                        | `2.0.0`          |
+| `opensearch-dashboards`  | [`v2.5.0`](https://github.com/opensearch-project/OpenSearch-Dashboards/releases/tag/2.5.0)             | `2.0.0`          |
+| `cerebro`                | [`v0.9.4`](https://github.com/lmenezes/cerebro/releases/tag/v0.9.4)                                    | `No update`      |
+| `logging-operator`       | [`v3.17.10`](https://github.com/banzaicloud/logging-operator/releases/tag/3.17.10)                     | `3.17.7`         |
+| `loki-distributed`       | [`v2.7.3`](https://github.com/grafana/loki/releases/tag/v2.7.3)                                        | `New package`    |
+| `minio-ha`               | [`vRELEASE.2023-01-12T02-06-16Z`](https://github.com/minio/minio/tree/RELEASE.2023-01-12T02-06-16Z)    | `New package`    |
+
+## Bug Fixes and Changes 🐛
+
+- Add support for Kubernetes 1.25.
```
Comment (ralgozino) [inline docs/releases/v3.1.0.md:23]:
```suggestion
- Added support for Kubernetes 1.25.
```
to be coherent with the other items' tense

File: docs/releases/v3.1.0.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+- Introduced Loki distributed package ready for production use.
+- Introduced minio-ha package to store errors from Fluentd, and as an S3 backend for Loki.
+- The Kubernetes ClusterFlow now exclude by default all the  infrastructural (KFD) logs and a new Infra ClusterFlow has been added to get all the infrastructural (KFD) logs.
```
Comment (ralgozino) [inline docs/releases/v3.1.0.md:26]:
```suggestion
- The Kubernetes ClusterFlow now exclude by default all the infrastructural (KFD) log, a new Infra ClusterFlow has been added to collect all the infrastructural (KFD) logs.
```

File: katalog/loki-distributed/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,1366 @@
+# Copyright (c) 2022 SIGHUP s.r.l All rights reserved.
```
Comment (ralgozino) [inline katalog/loki-distributed/deploy.yml:1]:
```suggestion
# Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
```

File: katalog/minio-ha/MAINTENANCE.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# MinIO HA - maintenance
+
+To maintain the MinIO package, you should follow this steps.
```
Comment (ralgozino) [inline katalog/minio-ha/MAINTENANCE.md:3]:
```suggestion
To maintain the MinIO package, you should follow these steps.
```

File: katalog/minio-ha/MAINTENANCE.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+```bash
+helm template minio-logging /tmp/minio/helm/minio --values MAINTENANCE.values.yaml -n logging > minio-built.yaml
+```
+
+What was customized (what differs from the helm template command):
+
+- Config has been moved from the template output and generated via kustomize
+- Add a custom init job to create buckets and add 7 day retention
+- Add `preferredDuringSchedulingIgnoredDuringExecution` on minio pods
```
Comment (ralgozino) [inline katalog/minio-ha/MAINTENANCE.md:18]:
```suggestion
- Added a custom init job to create buckets and add 7 day retention
- Added `preferredDuringSchedulingIgnoredDuringExecution` on minio pods
```

File: katalog/minio-ha/MAINTENANCE.values.yaml [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    # for node metrics
+    relabelConfigs: {}
+    # for cluster metrics
+    relabelConfigsCluster: {}
+      # metricRelabelings:
+      #   - regex: (server|pod)
+    #     action: labeldrop
+    # namespace: monitoring
+    # interval: 30s
+    # scrapeTimeout: 10s
```
Comment (ralgozino) [inline katalog/minio-ha/MAINTENANCE.values.yaml:112]: nitpick: missing new line at the end of the file.

File: katalog/minio-ha/README.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+MinIO is a popular distributed object storage system that allows organizations to deploy highly available
+and scalable storage infrastructure.
+In order to achieve high availability (HA) for MinIO, a cluster of multiple MinIO nodes must be deployed,
+with each node having access to its own set of disks.
+
+## Requirements
+
+- Kubernetes >= `1.23.0`
+- Kustomize >= `v3.5.3`
+- [prometheus-operator from KFD monitoring module][prometheus-operator]\
```
Comment (ralgozino) [inline katalog/minio-ha/README.md:14]:
```suggestion
- [prometheus-operator from KFD monitoring module][prometheus-operator]
```

File: katalog/minio-ha/README.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
+# MinIO HA
+
+<!-- <KFD-DOCS> -->
+
+MinIO is a popular distributed object storage system that allows organizations to deploy highly available
+and scalable storage infrastructure.
+In order to achieve high availability (HA) for MinIO, a cluster of multiple MinIO nodes must be deployed,
+with each node having access to its own set of disks.
```
Comment (ralgozino) [inline katalog/minio-ha/README.md:8]: disks or PVC?

File: katalog/minio-ha/README.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+## Image repository and tag
+
+* MinIO image: `minio/minio`
+* MinIO repo: [MinIO on Github][minio-gh]
+
+## Configuration
+
+MinIO HA is deployed in the following configuration:
+
+- Three Pod MinIO statefulset with 2 disks per Pod
```
Comment (ralgozino) [inline katalog/minio-ha/README.md:28]: disks or PVC?

File: katalog/minio-ha/initialize-minio-buckets.yaml [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+metadata:
+  name: minio-setup:wait-for-minio
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: wait-for-minio
+subjects:
+  - kind: ServiceAccount
+    name: wait-for-minio
+    namespace: logging
```
Comment (ralgozino) [inline katalog/minio-ha/initialize-minio-buckets.yaml:76]: nitpick: missing new line at the end of the file

File: katalog/minio-ha/kustomization.yaml [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    newTag: v1.6
+  - name: registry.sighup.io/fury/minio/mc
+    newTag: RELEASE.2023-01-28T20-29-38Z
+  - name: registry.sighup.io/fury/minio
+    newTag: RELEASE.2023-01-12T02-06-16Z
+
+secretGenerator:
+  - name: minio-logging
+    envs:
+      - config/minio.root.env
```
Comment (ralgozino) [inline katalog/minio-ha/kustomization.yaml:26]: nitpick: missing new line at the end of the file

**Review by ralgozino — CHANGES_REQUESTED** [review]
some minor cosmetic changes and suggestions

File: katalog/minio-ha/README.md (hunk 10 above)
Comment (nutellinoit) [reply katalog/minio-ha/README.md:8]:
```suggestion
with each node having access to its own set of PVCs.
```

File: katalog/loki-distributed/README.md [hunk 14]
```diff
@@ -25,16 +24,24 @@ It does not index the contents of the logs, but rather a set of labels for each
...
+    resources:
+      requests:
+        cpu: 100m
+        memory: 128Mi
+      limits:
+        cpu: 500m
+        memory: 1024Mi
+    ```
 
 ## Deployment
```
[the remark this answers is not in the document]
Comment (nutellinoit) [reply katalog/loki-distributed/README.md:42]: Done!

**Review by ralgozino — DISMISSED** [review]
LGTM 🚀

**Review by ralgozino — APPROVED** [review]

**Review by ralgozino — APPROVED** [review]

**Pull request merged**